File: test_data/python/test_main/aas_core_meta.v3/input/snippets/Verification/submodel_elements_have_identical_semantic_IDs.py
def submodel_elements_have_identical_semantic_ids(
    elements: Iterable[aas_types.SubmodelElement],
) -> bool:
    """
    Check that all :paramref:`elements` have the identical
    :py:attr:`.types.HasSemantics.semantic_id`.
    """
    that_semantic_id = None  # type: Optional[aas_types.Reference]

    for element in elements:
        if element.semantic_id is None:
            continue

        if that_semantic_id is None:
            that_semantic_id = element.semantic_id
            continue

        this_semantic_id = element.semantic_id

        if len(that_semantic_id.keys) != len(this_semantic_id.keys):
            return False

        for this_key, that_key in zip(this_semantic_id.keys, that_semantic_id.keys):
            if this_key.value != that_key.value:
                return False

    return True

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/submodel_elements_have_identical_semantic_IDs.java
/**
* Check that all elements have the identical
* {@link IHasSemantics#getSemanticId() semanticId}'s.
*/
public static boolean submodelElementsHaveIdenticalSemanticIds(
  Iterable<? extends ISubmodelElement> elements) {
  Objects.requireNonNull(elements);

  IReference thatSemanticId = null;

  for (ISubmodelElement element : elements) {
    if (!element.getSemanticId().isPresent()) {
      continue;
    }

    if (thatSemanticId == null) {
      thatSemanticId = element.getSemanticId().get();
      continue;
    }

    IReference thisSemanticId = element.getSemanticId().get();

    if (thatSemanticId.getKeys().size() != thisSemanticId.getKeys().size()) {
      return false;
    }

    for (int i = 0; i < thisSemanticId.getKeys().size(); i++) {
      if (!Objects.equals(thatSemanticId.getKeys().get(i).getValue(),
                          thisSemanticId.getKeys().get(i).getValue())) {
        return false;
      }
    }
  }

  return true;
}
